app/modules/auth/auth_router.py

```
@@ -52,52 +633,527 @@ async def login(login_request: LoginRequest):
+                )
+
+        except Exception as e:
+            return JSONResponse(
+                content={"error": f"Failed to unlink provider: {str(e)}"},
+                status_code=400,
+            )
+
+    @auth_router.get("/account/check-email")
+    async def check_email_providers(
+        email: str,
+        db: Session = Depends(get_db),
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)